source/modes/window.c

```diff
@@ -1043,20 +1044,21 @@ static cairo_surface_t *get_net_wm_icon(xcb_window_t xid,
   return surface;
 }
 static cairo_surface_t *_get_icon(const Mode *sw, unsigned int selected_line,
-                                  unsigned int size) {
+                                  unsigned int size, guint scale) {
   WindowModePrivateData *rmpd = mode_get_private_data(sw);
   client *c = window_client(rmpd, rmpd->ids->array[selected_line]);
   if (c == NULL) {
     return NULL;
   }
-  if (c->icon_fetch_size != size) {
+  if (c->icon_fetch_size != size || c->icon_fetch_scale != scale) {
     if (c->icon) {
       cairo_surface_destroy(c->icon);
       c->icon = NULL;
     }
     c->thumbnail_checked = FALSE;
     c->icon_checked = FALSE;
   }
+  // TODO: apply scaling to the following two routines
```

Comment:
@MoetaYuko is this actually required? Also, this is the xcb window mode. Was a fix needed here as well? Maybe it can be brought upstream.

Reply:
TBH I don't know their usecases so I marked them to avoid potential ignorance. Can be deleted if you think it's fine to leave them as-is.

Reply:
Ok I see. I'll keep anyway for now and make a release as there are a few useful fixes in the main branch since the latest one.